homeassistant/components/aprilaire/sensor.py

```diff
@@ -0,0 +1,309 @@
+        status_map=DEHUMIDIFICATION_STATUS_MAP,
+        options=list(set(DEHUMIDIFICATION_STATUS_MAP.values())),
+    ),
+    AprilaireStatusSensorDescription(
+        key="humidification_status",
+        translation_key="humidification_status",
+        device_class=SensorDeviceClass.ENUM,
+        status_key=Attribute.HUMIDIFICATION_AVAILABLE,
+        value_key=Attribute.HUMIDIFICATION_STATUS,
+        status_map=HUMIDIFICATION_STATUS_MAP,
+        options=list(set(HUMIDIFICATION_STATUS_MAP.values())),
+    ),
+    AprilaireStatusSensorDescription(
+        key="ventilation_status",
+        translation_key="ventilation_status",
+        device_class=SensorDeviceClass.ENUM,
+        status_key=Attribute.VENTILATION_AVAILABLE,
+        value_key=Attribute.VENTILATION_STATUS,
+        status_map=VENTILATION_STATUS_MAP,
+        options=list(set(VENTILATION_STATUS_MAP.values())),
+    ),
+    AprilaireStatusSensorDescription(
+        key="air_cleaning_status",
+        translation_key="air_cleaning_status",
+        device_class=SensorDeviceClass.ENUM,
+        status_key=Attribute.AIR_CLEANING_AVAILABLE,
+        value_key=Attribute.AIR_CLEANING_STATUS,
+        status_map=AIR_CLEANING_STATUS_MAP,
+        options=list(set(AIR_CLEANING_STATUS_MAP.values())),
+    ),
+    AprilaireStatusSensorDescription(
+        key="fan_status",
+        translation_key="fan_status",
+        device_class=SensorDeviceClass.ENUM,
+        status_key=None,
+        value_key=Attribute.FAN_STATUS,
+        status_map=FAN_STATUS_MAP,
+        options=list(set(FAN_STATUS_MAP.values())),
+    ),
+)
+
+
+class BaseAprilaireSensor(BaseAprilaireEntity, SensorEntity):
+    """Base sensor entity for Aprilaire."""
+
+    entity_description: AprilaireSensorDescription
+    status_sensor_available_value: int | None = None
+    status_sensor_exists_values: list[int] | None = None
+
+    def __init__(
+        self,
+        coordinator: AprilaireCoordinator,
+        description: AprilaireSensorDescription,
+        unique_id: str,
+    ) -> None:
+        """Initialize a sensor for an Aprilaire device."""
+
+        self.entity_description = description
+
+        super().__init__(coordinator, unique_id)
+
+    @property
+    def exists(self) -> bool:
+        """Return True if the sensor exists."""
+
+        if (
+            self.entity_description.status_key is None
+            or self.status_sensor_exists_values is None
+        ):
+            return True
+
+        return (
+            self.coordinator.data.get(self.entity_description.status_key)
+            in self.status_sensor_exists_values
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return True if the sensor is available."""
+
+        if (
+            self.entity_description.status_key is None
+            or self.status_sensor_available_value is None
+        ):
+            return True
+
+        if not super().available:
+            return False
```

Comment:
Shouldn't this case be first? Or should the fan status sensor, which is the only sensor which doesn't have a status_key, never be unavailable?

Reply:
@emontnemery Yes that is correct. The fan status sensor is always available.